src/Apps/Order2/Routes/Checkout/Hooks/useCheckoutTracking.ts

```diff
@@ -2,135 +2,153 @@ import {
   ActionType,
   type ClickedCancelExpressCheckout,
   type ClickedExpressCheckout,
+  ContextModule,
   type ErrorMessageViewed,
   type ExpressCheckoutViewed,
   type OwnerType,
+  type SubmittedOffer,
   type SubmittedOrder,
 } from "@artsy/cohesion"
+import { useOrder2Tracking } from "Apps/Order2/Hooks/useOrder2Tracking"
 import { useAnalyticsContext } from "System/Hooks/useAnalyticsContext"
 import { useMemo } from "react"
 import { useTracking } from "react-tracking"
 
-export const useCheckoutTracking = () => {
+export const useCheckoutTracking = ({
+  source,
+  mode,
+}: {
+  source: string
+  mode: "BUY" | "OFFER" | string
+}) => {
   const { trackEvent } = useTracking()
+  const order2Tracking = useOrder2Tracking()
   const analytics = useAnalyticsContext()
   const contextPageOwnerId = analytics.contextPageOwnerId as string
   const contextPageOwnerSlug = analytics.contextPageOwnerSlug as string
   const contextPageOwnerType = analytics.contextPageOwnerType as OwnerType
+  const flow = buyOrOfferValue(
+    mode,
+    source === "PARTNER_OFFER" ? "Partner offer" : "Buy now",
+    "Make offer",
+  )
 
-  const trackingCalls = useMemo(() => {
+  const checkoutTracking = useMemo(() => {
     return {
-      submittedOrder: ({
-        source,
-        walletType,
-      }: {
-        source: string
-        walletType?: string
-      }) => {
-        const payload: SubmittedOrder = {
-          action: ActionType.submittedOrder,
+      clickedExpressCheckout: ({ walletType }: { walletType: string }) => {
+        const payload: ClickedExpressCheckout = {
+          action: ActionType.clickedExpressCheckout,
           context_page_owner_type: contextPageOwnerType,
-          order_id: contextPageOwnerId,
-          flow: source === "PARTNER_OFFER" ? "Partner offer" : "Buy now",
-          ...(walletType ? { credit_card_wallet_type: walletType } : {}),
-        }
-
-        trackEvent(payload)
-      },
-
-      errorMessageViewed: ({
-        error_code,
-        title,
-        message,
-        flow,
-      }: {
-        error_code: string | null
-        title: string
-        message: string
-        flow: string
-      }) => {
-        const payload: ErrorMessageViewed = {
-          action: ActionType.errorMessageViewed,
-          context_owner_type: contextPageOwnerType,
-          context_owner_id: contextPageOwnerId,
-          title,
-          message,
-          error_code: error_code || undefined,
+          context_page_owner_id: contextPageOwnerId,
           flow,
+          credit_card_wallet_type: walletType,
         }
 
         trackEvent(payload)
       },
-
-      expressCheckoutViewed: ({
-        order,
-        walletType,
-      }: {
-        order: { source: string; mode: string }
-        walletType: string[]
-      }) => {
+      expressCheckoutViewed: ({ walletType }: { walletType: string[] }) => {
```

Comment:
nitpick: I see that here the prop `walletType` type is `string[]`, but in the following function, the `walletType` type is `string`, should we update the naming to `walletTypes` instead?